test/helpers/kubectl.go

```diff
@@ -420,9 +437,81 @@ func (kub *Kubectl) CiliumReport(namespace string, pod string, commands []string
 		fmt.Fprintln(wr, out.CombineOutput())
 	}
 	fmt.Fprint(wr, "StackTrace Ends\n")
+	kub.CiliumReportFull(namespace, pod)
 	return nil
 }
 
+//CiliumReportFull writes to testResultsPath all the output of a variety of
+//cilium commands to know the state of cilium
+func (kub *Kubectl) CiliumReportFull(namespace string, pod string) {
+	reportCiliumCommands := map[string]string{
+		"cilium endpoint list -o json": "endpoint_list_txt",
+		"cilium service list -o json":  "service_list.txt",
+		"cilium config":                "config.txt",
+		"cilium bpf lb list":           "bpf_lb_list.txt",
+		"cilium bpf ct list global":    "bpf_ct_list.txt",
+		"cilium bpf tunnel list":       "bpf_tunnel_list.txt",
+		"cilium policy get":            "policy_get.txt",
+		"cilium status":                "status.txt",
+	}
+	reportEndpointCommands := map[string]string{
+		"cilium endpoint get %s":    "endpoint_get_%s.txt",
+		"cilium bpf policy list %s": "bpf_policy_list_%s.txt",
+	}
+
+	testPath, err := ReportDirectory()
+	if err != nil {
+		kub.logCxt.Errorf("Can't create Test results path '%s' err '%s'", testPath, err)
+	}
+
+	reportCmds := map[string]string{}
+	for cmd, logfile := range reportCiliumCommands {
+		command := fmt.Sprintf("kubectl exec -n %s %s -- %s", namespace, pod, cmd)
+		reportCmds[command] = logfile
+	}
+	reportMap(testPath, reportCmds, kub.Node)
+
+	for _, ep := range kub.CiliumEndpointsIDs(pod) {
+		for cmd, logfile := range reportEndpointCommands {
+			command := fmt.Sprintf(cmd, ep)
+			res := kub.Node.Exec(fmt.Sprintf(
+				"kubectl exec -n %s %s -- %s", namespace, pod, command))
+			err = ioutil.WriteFile(
+				fmt.Sprintf("%s/%s", testPath, fmt.Sprintf(logfile, ep)),
+				res.CombineOutput().Bytes(),
+				0777)
+			if err != nil {
+				kub.logCxt.Errorf("Can not create test results for command '%s' err='%s'",
+					command, err)
+			}
+		}
+	}
+
+	for _, id := range kub.CiliumEndpointsIdentityIDs(pod) {
+		cmd := fmt.Sprintf("kubectl exec -n %s %s -- cilium identity get %s", namespace, pod, id)
+
+		res := kub.Node.Exec(cmd)
+		err = ioutil.WriteFile(
+			fmt.Sprintf("%s/%s", testPath, fmt.Sprintf("identity_%s.txt", id)),
+			res.CombineOutput().Bytes(),
+			0777)
+		if err != nil {
+			kub.logCxt.Errorf("Can not create test results for command '%s' err='%s'", cmd, err)
+		}
+	}
+}
+
+//ReportEnd Write to the testReport directory all logs related with kubernetes status
```

Comment:
Incorrect grammar; "ReportEnd dumps the output of a variety of Kubernetes commands to directory testReport".
The name of this function isn't informative either. I'm not able to understand that it gets the output of Kubernetes commands from its name. What about `DumpKubernetesCommands`?